tests/DocBlock/TypeExpressionTest.php

```diff
@@ -654,4 +666,71 @@ public static function provideSortTypesCases(): iterable
             '-8.2023437675747321e-18_446_744_073_709_551_616|18_446_744_073_709_551_616',
         ];
     }
+
+    /**
+     * @return list<array{int, string}|list>
+     */
+    private function checkInnerTypeExpressionsStartIndex(TypeExpression $typeExpression): array
+    {
+        $innerTypeExpressions = \Closure::bind(fn () => $typeExpression->innerTypeExpressions, null, TypeExpression::class)();
+
+        $res = [];
+        foreach ($innerTypeExpressions as ['start_index' => $innerStartIndex, 'expression' => $innerExpression]) {
+            $innerExpressionStr = $innerExpression->toString();
+            self::assertSame(
+                $innerExpressionStr,
+                substr($typeExpression->toString(), $innerStartIndex, \strlen($innerExpressionStr))
+            );
+
+            $res[] = [$innerStartIndex, $innerExpressionStr];
+
+            $res[] = $this->checkInnerTypeExpressionsStartIndex($innerExpression);
+        }
+
+        return $res;
+    }
+
+    private function clearPcreRegexCache(): void
+    {
+        // there is no explicit php function to clear PCRE regex cache, but based
+        // on https://www.php.net/manual/en/intro.pcre.php there are 4096 cache slots
+        // pruned in FIFO fashion, so to clear the cache, replace all existing
+        // cache slots with dummy regexes
+        for ($i = 0; $i < 4096; ++$i) {
+            preg_match('/^'.$i.'/', '');
+        }
+    }
+
+    /**
+     * Parse type expression with and without PCRE JIT.
```

Comment:
I am missing the description of the purpose of this, why it's needed and when we can get rid of it. Would that even be possible?

also, manually clearing the cache is fishy too. I understand that we need to overcome the issue, but I wonder what will be the clean setup that we don't need it.
(e.g. things like "revert this PR after PHP 8.6 is lowest supported version" etc)

Reply:
It ensures that results are the same for both scenarios - see #6997 for a root cause. Probably we can get rid of it when upstream issue is fixed, but on the other hand it should catch any other difference between JIT/non-JIT usage.

Reply:
Feel free to read https://github.com/PCRE2Project/pcre2/issues/261#issuecomment-1578992003 and link that issue in the method description, as this method asserts very critical part of the Fixer, I personally prefer to keep this testing forever.

Faster PCRE regex cache clearing is discussed in https://github.com/php/php-src/issues/11374#issuecomment-1578954380, feel free to add the link in the comment as well. Contribution to php-src to clear the cache on `pcre.jit` php.ini change is welcomed and IMHO it can even be classified as php-src bugfix, ie. it can go to PHP 8.1.x.

Reply:
John Doe, opening the codebase in 2025, will not have #6997 awareness when looking into this code.
and I hope that in 2024 we can remove this hack, and instead expect that php/module we need is vx.y.z+

Reply:
@keradus I believe Git history is exactly for that, unless you assume Github won't work in 2025 and John Doe can't go to the issue mentioned in the commit message 😉.

Reply:
git history will be tricky with project changing formatting, ordering, files structure etc again and again ;) also, one-two liner next to non-intuitive code is better than asking developer "try to find the background of this non-intuitive code in history of git, maybe in one of the PR comments"

Reply:
Yes, it is good practice to explicitly mention the links and possible removal date/condition in the code.

The JIT/non-JIT parse in the tests should be kept forever to prevent related bugs, but the "4096 FIFO cleanup" should be possible to be removed for PHP 8.3+ as I expect https://github.com/php/php-src/pull/11396 to be merged soon. @keradus can you please document this in the code? Also, can you please publish a patch release as this bugfix PR can be important for a lot of people?

Reply:
I have stashed a comment localy, I will watch the php-src PR and if it will be merged, I will then land a comment via another PR.